File: app/src/main/java/com/gmail/mostafa/ma/saleh/yawa/models/Day.java
package com.gmail.mostafa.ma.saleh.yawa.models;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

public class Day implements Serializable {
    @SerializedName("dt")
    @Expose
    public int dt;
    @SerializedName("temp")
    @Expose
    public Temp temp;
    @SerializedName("pressure")
    @Expose
    public double pressure;
    @SerializedName("humidity")
    @Expose
    public int humidity;
    @SerializedName("weather")
    @Expose
    public Weather[] weather;
    @SerializedName("speed")
    @Expose
    public float speed;
    @SerializedName("deg")
    @Expose
    public float deg;
    @SerializedName("clouds")
    @Expose
    public float clouds;

    public class Weather implements Serializable {

        @SerializedName("id")
        @Expose
        public int id;
        @SerializedName("main")
        @Expose
        public String main;
        @SerializedName("description")
        @Expose
        public String description;
        @SerializedName("icon")
        @Expose
        public String icon;
    }

    public class Temp implements Serializable {

        @SerializedName("day")
        @Expose
        public double day;
        @SerializedName("min")
        @Expose
        public double min;
        @SerializedName("max")
        @Expose
        public double max;
        @SerializedName("night")
        @Expose
        public double night;
        @SerializedName("eve")
        @Expose
        public double eve;
        @SerializedName("morn")
        @Expose
        public double morn;
    }
}

File: app/src/main/java/com/gmail/mostafa/ma/saleh/yawa/models/Day.kt
package com.gmail.mostafa.ma.saleh.yawa.models

import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize

@Parcelize
data class Day(@SerializedName("date")
               @Expose
               val date: Int,
               @SerializedName("temp")
               @Expose
               val temperature: Temperature,
               @SerializedName("pressure")
               @Expose
               val pressure: Double,
               @SerializedName("humidity")
               @Expose
               val humidity: Int,
               @SerializedName("weather")
               @Expose
               val weather: List<Weather>,
               @SerializedName("windSpeed")
               @Expose
               val windSpeed: Double,
               @SerializedName("windDirection")
               @Expose
               val windDirection: Double,
               @SerializedName("clouds")
               @Expose
               val clouds: Double) : Parcelable {

    @Parcelize
    data class Weather(@SerializedName("id")
                       @Expose
                       val id: Int,
                       @SerializedName("main")
                       @Expose
                       val main: String,
                       @SerializedName("description")
                       @Expose
                       val description: String,
                       @SerializedName("icon")
                       @Expose
                       val icon: String) : Parcelable

    @Parcelize
    data class Temperature(@SerializedName("day")
                           @Expose
                           val day: Double,
                           @SerializedName("min")
                           @Expose
                           val min: Double,
                           @SerializedName("max")
                           @Expose
                           val max: Double,
                           @SerializedName("night")
                           @Expose
                           val night: Double,
                           @SerializedName("eve")
                           @Expose
                           val eve: Double,
                           @SerializedName("morn")
                           @Expose
                           val morn: Double) : Parcelable
}
